src/core/BufferAttribute.d.ts

```diff
@@ -151,7 +151,7 @@ export class Float64Attribute extends BufferAttribute {
 export class Int8BufferAttribute extends BufferAttribute {
 
 	constructor(
-		array: Iterable<number> | ArrayLike<number> | ArrayBuffer,
+		array: Iterable<number> | ArrayLike<number> | ArrayBuffer | number,
```

Comment:
These `array` are passed to typed array constructors, which accept a number as the length of the array to be constructed.

Reply:
That is true but `number` does not represent the semantics of an array. In other words: It's intended that users pass in an array when creating a concrete buffer attribute.

Reply:
That usage was found in some examples but I agree with you. Reverted.

Reply:
> That usage was found in some examples but I agree with you.

Can you please link here these examples?^^

Reply:
@Mugen87 

* https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/MMDLoader.js#L758
* https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/MMDLoader.js#L758
* https://github.com/mrdoob/three.js/blob/dev/examples/webgl_custom_attributes_lines.html#L115
* https://github.com/mrdoob/three.js/blob/dev/src/core/BufferGeometry.js#L290
* https://github.com/mrdoob/three.js/blob/dev/src/core/BufferGeometry.js#L298
* https://github.com/mrdoob/three.js/blob/dev/src/helpers/FaceNormalsHelper.js#L49
* https://github.com/mrdoob/three.js/blob/dev/src/helpers/VertexNormalsHelper.js#L48

Reply:
Thanks for the listing. Okay, that's more than I've expected...

Um, in this case it's okay to add the additional typing. There are currently no docs about these concrete buffer attributes that would tell otherwise.

Sorry, but could you restore your proposal in an other PR? 😇

Reply:
I think it can be fixed by changing these `new BufferAttribute(num)` to `new BufferAttribute(new TypedArray(num))` or by changing the definitions, either will work.

I make a PR at mrdoob/three.js#17717 and leave the decision to you.